cluster/juju/prereqs/ubuntu-juju.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Copyright 2014 Canonical LTD. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# If you find any bugs within this script - please file bugs against the
+# Kubernetes Juju Charms project - located here: https://github.com/whitmo/bundle-kubernetes
+
+function check_for_ppa(){
+    grep -s ^ /etc/apt/sources.list /etc/apt/sources.list.d/* | grep juju
+}
+
+function package_status(){
+    local pkgstatus=`dpkg-query -W --showformat='${Status}\n' $1|grep "install ok installed"`
```

Comment:
Please use `$(...)` syntax for command substitution.

There's a problem mixing `local` and `$(...)` in that the `local` will "hide" the error state of the command if it fails. Split it in two lines instead.

You also need double quotes around variables that are supposed to expand to a single name, such as `"$1"`. Probably better to assign that to another variable, e.g. `${pkgname}`, make sure you keep it quoted everywhere since it's always supposed to be a single word.

Instead of using `| grep` I think you can just check that the string matches exactly.

Also, always use `[[ ... ]]` for comparisons since it's a bash built-in with better support for words and nicer matching.

```shell
  local pkgname=$1
  local pkgstatus
  pkgstatus=$(dpkg-query -W --showformat='${Status}\n' "${pkgname}"`
  if [[ "${pkgstatus}" != "install ok installed" ]]; then
    echo "Missing package ${pkgname}"
    sudo apt-get --force-yes --yes install "${pkgname}"
  fi
```